demos/LowAllocationWebServer/HttpHeaders.cs

```diff
@@ -0,0 +1,129 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using System.Text.Utf8;
+using LowAllocationServer;
+
+namespace System.Net.Http.Buffered
+{
+    public struct HttpHeaders : IEnumerable<KeyValuePair<Utf8String, Utf8String>>
+    {
+        private readonly ByteSpan _bytes;
+        
+        public HttpHeaders(ByteSpan bytes)
+        {
+            _bytes = bytes;            
+        }                
+
+        public Utf8String this[string headerName]
+        {
+            get
+            {
+                foreach (var header in this)
+                {
+                    if (header.Key == headerName)
+                    {
+                        return header.Value;
+                    }
+                }
+
+                return Utf8String.Empty;
```

Comment:
Given that we don't cache the values, I worry that if we return Utf8String? people will do two expensive lookups, one to check for null and one to get the value. But throwing would mean we need to add Try versions.